### PR TITLE
Cancel mission upload fix

### PR DESCRIPTION
**Pull request by unipheas**
I started from master because it seemed to be the most updated and the backend worked. From that I added a new 'Cancel Upload' button to the UI. The views that contained the buttons were correctly setup with constraints so it works on all devices and stays in their corners. I also connected it to MapViewController.swift and setup and alert controller that would confirm cancelation.

### CODE REVIEW

File: Sources/DronecodeSDKSwiftDemo/Controllers/MapViewController.swift [hunk 1]
```diff
@@ -10,6 +10,7 @@ class MapViewController: UIViewController, CLLocationManagerDelegate {
     @IBOutlet weak var mapView: MKMapView!
     @IBOutlet weak var feedbackLabel: UILabel!
     @IBOutlet weak var uploadMissionButton: UIButton!
+	@IBOutlet var cancelMissionUploadButton: UIButton!
```
Comment (julianoes) [inline Sources/DronecodeSDKSwiftDemo/Controllers/MapViewController.swift:13]: Indenting?

File: Sources/DronecodeSDKSwiftDemo/Controllers/MapViewController.swift [hunk 2]
```diff
@@ -44,22 +45,23 @@ class MapViewController: UIViewController, CLLocationManagerDelegate {
...
-        startMissionButton.layer.cornerRadius    = UI_CORNER_RADIUS_BUTTONS
-        createFlightPathButton.layer.cornerRadius    = UI_CORNER_RADIUS_BUTTONS
-        centerMapOnDroneButton.layer.cornerRadius    = UI_CORNER_RADIUS_BUTTONS
-        pauseMissionButton.layer.cornerRadius    = UI_CORNER_RADIUS_BUTTONS
-        setCurrentIndexButton.layer.cornerRadius    = UI_CORNER_RADIUS_BUTTONS
-        downloadMissionButton.layer.cornerRadius    = UI_CORNER_RADIUS_BUTTONS
-        isMissionFinishedButton.layer.cornerRadius    = UI_CORNER_RADIUS_BUTTONS
+        uploadMissionButton.layer.cornerRadius   		= UI_CORNER_RADIUS_BUTTONS
+        startMissionButton.layer.cornerRadius    		= UI_CORNER_RADIUS_BUTTONS
+		cancelMissionUploadButton.layer.cornerRadius 	= UI_CORNER_RADIUS_BUTTONS
```
Comment (julianoes) [inline Sources/DronecodeSDKSwiftDemo/Controllers/MapViewController.swift:50]: This looks odd on GitHub. Do you use the same spaces/tabs settings as the existing project?

File: Sources/DronecodeSDKSwiftDemo/Controllers/MapViewController.swift [hunk 3]
```diff
@@ -147,6 +149,26 @@ class MapViewController: UIViewController, CLLocationManagerDelegate {
             .subscribe()
             .disposed(by: disposeBag)
     }
+	
+	@IBAction func cancelMissionUploadPressed(_ sender: Any) {
+		displayFeedback(message: "Cancel Upload Pressed")
+		
+		let ac = UIAlertController(title: "Cancel Upload", message: "Are you sure?", preferredStyle: .alert)
```
Comment (julianoes) [inline Sources/DronecodeSDKSwiftDemo/Controllers/MapViewController.swift:156]: Nice!

**Review by julianoes — APPROVED** [review]
Thanks! Just some nitpicks.

File: Sources/DronecodeSDKSwiftDemo/Controllers/MapViewController.swift (hunk 1 above)
Comment (unipheas) [reply Sources/DronecodeSDKSwiftDemo/Controllers/MapViewController.swift:13]: This was missed and I see I didn't conform to what was previously done by using weak.

File: Sources/DronecodeSDKSwiftDemo/Controllers/MapViewController.swift (hunk 2 above)
Comment (unipheas) [reply Sources/DronecodeSDKSwiftDemo/Controllers/MapViewController.swift:50]: This was something I did to clean up the code. It can be corrected.

**Comment by JonasVautherin** [comment]
I am testing it right now, though I don't think it compiles with SDK `0.3.0` (as defined in the `Cartfile`). Did you update to `0.4.1`? Trying it now :-)

**Comment by unipheas** [comment]
> I am testing it right now, though I don't think it compiles with SDK `0.3.0` (as defined in the `Cartfile`). Did you update to `0.4.1`? Trying it now :-)

I did not update and it was able to compile. I'm updating now.

**Comment by JonasVautherin** [comment]
Working for me, but I had to bump the Cartfile to `0.4.1`, and make sure that Cartfile.resolved had grpc-swift set to `0.8.0` because of [this bug](https://github.com/Dronecode/DronecodeSDK-Swift/pull/122).

Maybe it's worth changing the Cartfile here to `0.4.1` then, what do you think?

**Comment by JonasVautherin** [comment]
Well let's merge that, I'll update the Cartfile later.

Thanks a lot @unipheas! Feel free to edit other parts of the example app if you want to, I think it is super nice for people to get started :-)

**Comment by unipheas** [comment]
> Well let's merge that, I'll update the Cartfile later.
> 
> Thanks a lot @unipheas! Feel free to edit other parts of the example app if you want to, I think it is super nice for people to get started :-)

Sorry, I had to go to bed. Thanks for merging and I will definitely continue working on the example app or anything else I can to help.

**Pull request merged**